File: api/routes/ticker_list.py
from flask import Blueprint, jsonify
from api.routes.utils import load_cached_data

ticker_list = Blueprint('ticker_list', __name__)

@ticker_list.route('/ticker-list', methods=['GET'])
def handle_ticker_list():
    cached_data = load_cached_data()
    tickers = [{'label': currency['symbol'], 'id': index} for index, currency in enumerate(cached_data['data'], start=1)]
    return jsonify(tickers)

File: internal/handlers/ticker_list.go
package handlers

import (
	"net/http"

	"github.com/gin-gonic/gin"
)

func FetchTickerList(BaseURL string) gin.HandlerFunc {
	return func(c *gin.Context) {
		// Load cached data or fetch from API
		cachedData, err := loadCachedData(BaseURL)
		if err != nil {
			c.JSON(http.StatusInternalServerError, gin.H{"error": "Failed to load data"})
			return
		}

		// Extract tickers from cached data
		var tickers []gin.H
		for index, currency := range cachedData.Data {
			tickers = append(tickers, gin.H{
				"label": currency.Symbol,
				"id":    index + 1,
			})
		}

		c.JSON(http.StatusOK, tickers)
	}
}

func loadCachedData(BaseURL string) (*CachedData, error) {
	// Implement the logic to load cached data or fetch from API
	// Use BaseURL to construct the full API endpoint
	// For now, return a placeholder
	return &CachedData{}, nil
}

type CachedData struct {
	Data []struct {
		Symbol string `json:"symbol"`
	} `json:"data"`
}
